clients/src/main/java/org/apache/kafka/clients/producer/MockProducer.java

```diff
@@ -72,6 +72,16 @@
     private long commitCount = 0L;
     private Map<MetricName, Metric> mockMetrics;
 
+    public RuntimeException initTransactionException = null;
```

Comment:
This allow us to simplify `RecordCollectorImplTest` significantly -- we can register a exception that we want to throw on the next invocation of the corresponding method.